R/DBConnection.R

```diff
@@ -106,6 +106,41 @@ setGeneric("dbSendQuery",
   valueClass = "DBIResult"
 )
 
+#' @export
+dbBreak <- structure(list(), class = "dbiAbort")
```

Comment:
I've looked over `readr` and love it. I think we should definitely adopt that API here, I'm working on a rough adaptation for us to go over. *But* if we do this, I'm not sure that the breaking from the `chunked` stuff should be same construct as the breaking from other DBI stuff.

In particular, the pattern used in `readr` is simply: if the callback returns `FALSE`, stop the iteration. What do you think @krlmlr, @jcheng5, should we go with this or something more explicit?